File: 5th semester/MA (Mobile Apps)/labs/2-7/server/model/Product.py
class Product:
    def __init__(self, pid, brand, gender, type, price, image, in_cart, quantity):
        self.pid = pid
        self.brand = brand
        self.gender = gender
        self.type = type
        self.price = price
        self.image = image
        self.in_cart = in_cart
        self.quantity = quantity

    def __repr__(self):
        return "Product {}=(brand= {}, gender= {}, type= {}, price= {}, image= '{}', in_cart= {}, quantity= {})".format(
            self.pid,
            self.brand,
            self.gender,
            self.type,
            self.price,
            self.image,
            self.in_cart,
            self.quantity,
        )

    def __str__(self):
        return "Product {}: brand= {}, gender= {}, type= {}, price= {}, image= '{}', in_cart= {}, quantity= {}".format(
            self.pid,
            self.brand,
            self.gender,
            self.type,
            self.price,
            self.image,
            self.in_cart,
            self.quantity,
        )

    @property
    def json(self):
        return {
            'pid': self.pid,
            'brand': self.brand,
            'gender': self.gender,
            'type': self.type,
            'price': self.price,
            'image': self.image,
            'in_cart': self.in_cart,
            'quantity': self.quantity,
        }

File: V/MA (Mobile Apps)/labs/2-7/non-native/footwear_shop/lib/model/Product.dart


class Product {
   int pid;
   String brand;
   String gender;
   String type;
   int price;
   String image;
   int inCart;
   int quantity;

   Product({this.pid, this.brand, this.gender, this.type, this.price, this.image, this.inCart, this.quantity});

   int subtotal() => price * quantity;

   Map<String, dynamic> toMap() {
      return {
        'id': pid,
        'brand': brand,
        'gender': gender,
        'type': type,
        'price': price,
        'image': image,
        'in_cart': inCart,
        'quantity': quantity
      };
   }

   factory Product.fromJson(Map<String, dynamic> json) {
     return Product(
       pid: json['pid'] as int,
       brand: json['brand'] as String,
       gender: json['gender'] as String,
       type: json['type'] as String,
       price: json['price'] as int,
       image: json['image'] as String,
       inCart: json['in_cart'] as int,
       quantity: json['quantity'] as int,
     );
   }

   @override
   bool operator ==(Object other) =>
      identical(this, other) ||
      other is Product &&
          runtimeType == other.runtimeType &&
          brand == other.brand &&
          gender == other.gender &&
          type == other.type &&
          price == other.price &&
          image == other.image;

   @override
   int get hashCode =>
      brand.hashCode ^
      gender.hashCode ^
      type.hashCode ^
      price.hashCode ^
      image.hashCode;

   @override
   String toString() {
     return 'Product{id: $pid, brand: $brand, gender: $gender, type: $type, price: $price, image: $image, inCart: $inCart, quantity: $quantity}';
   }
}
